soh/src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -112,14 +112,23 @@ void EnHeishi1_Init(Actor* thisx, GlobalContext* globalCtx) {
         }
     }
 
+    // eventChkInf[4] & 1 = Got Zelda's Letter
+    // eventChkInf[5] & 0x200 = Got item from impa
+    // eventChkInf[8] & 1 = Ocarina thrown in moat
+    bool metZelda = (gSaveContext.eventChkInf[4] & 1) && (gSaveContext.eventChkInf[5] & 0x200);
```

Comment:
do we need to check both of these?

Reply:
I check for both just in the off-chance that someone will only get zelda's letter and not get impa's reward. I guess we could just check for only Impa though, I just added it to make sure.

If I change this to only impa, should I just get rid of the bool entirely and check for the flag in the if statement directly?

Reply:
i'm not sure, i'm thinking about the future where we'll be able to exclude checks and wondering if disabling the impa's check might be a thing, but not the zelda's letter check?

i think for now checking for both is fine

Reply:
It doesn't really hurt checking for both either way for now. I'll keep it like it is.